docs/python/generators/examples_helpers.py

```
@@ -86,6 +86,116 @@ def _find_unquoted_word_on_nontrivial_line(word: str, text: str) -> bool:
+    # Canonical strong patterns first (e.g. macro invocations have priority over
+    # generic mentions of the same symbol).
+    for p in (strong_patterns or []):
+        if not _has_aforeach and any(x in p for x in ('AForEachUI', 'AUI_DECLARATIVE_FOR')):
+            continue
+        for m in re.finditer(p, text):
+            if _is_in_double_quotes(text, m.start()):
+                continue
+            pos = m.start()
+            cum = 0
+            line_idx = 0
+            for i, l in enumerate(lines):
```

Ruff (0.15.21)

[error] Ambiguous variable name: `l`

(E741)